src/io_header.c

```diff
@@ -338,10 +270,15 @@ void fill_MQ1_single_fields(MQ1_fields *mq1_field,
   mq1_field->colour_mode[index]     = mq1_h.colour_mode;
   mq1_field->gain_mode[index]       = mq1_h.gain_mode;
   /*threshold is float[8]*/
-  for (unsigned int i = 0; i < MQ1_FLOAT_LEN_THRESHOLD; ++i) {
-    mq1_field->threshold[index * MQ1_FLOAT_LEN_THRESHOLD + i] =
-      mq1_h.threshold[i];
-  }
+  unsigned int i               = -1;
+  mq1_field->threshold0[index] = mq1_h.threshold[++i];
+  mq1_field->threshold1[index] = mq1_h.threshold[++i];
+  mq1_field->threshold2[index] = mq1_h.threshold[++i];
+  mq1_field->threshold3[index] = mq1_h.threshold[++i];
+  mq1_field->threshold4[index] = mq1_h.threshold[++i];
+  mq1_field->threshold5[index] = mq1_h.threshold[++i];
+  mq1_field->threshold6[index] = mq1_h.threshold[++i];
+  mq1_field->threshold7[index] = mq1_h.threshold[++i];
```

Comment:
Can we hard-code 0, 1, 2... instead of `++i`? I know the idea is to be a bit flexible if the number of fields in threshold changes, but people still need to count the number of lines. Ideally we can use `MQ1_FLOAT_LEN_THRESHOLD` but seems no easy way. Same for the quad case.